include/wx/generic/maskededit.h

```diff
@@ -0,0 +1,681 @@
+      <    Convert all following characters to lower case.
+      ^    Switch off upper/lower case conversion.
+      \    Escape the next character and use it as a literal.
+     Notice the above specifications (except 'C' and 'c') are tested using
+     wxIsxxx functions. So, they are locale dependent.
+     Any other char is a literal. It is shown in the control, but it can
+     not be edited.
+     Literals are also used as fields separators.
+     '|'  is a special literal. It's invisible, takes no space, but still
+     is a fields separator.
+
+     Space is not a character (except for C and c commands).
+    */
+
+    bool SetMask(const wxString& mask);
+
+    //Set the colours for different cases.
+    void SetMaskedColours(const wxMaskedEditColours& colours);
+
+    //Set the flags for a field.
+    void SetFieldFlags(size_t index, const wxEditFieldFlags& flags)
+    {
+        m_params.fieldsFlags.at(index) = flags;
+    }
+
+    //Set all fields with the same flags.
+    void SetAllFieldsFlags(const wxEditFieldFlags& flags);
+
+    //Set the test function for a field.
+    void SetFieldFunction(size_t index, wxMaskedFieldFunc* fn, void* param);
+
+    //Set all fields with the same functions.
+    void SetAllFieldsFunction(wxMaskedFieldFunc* fn, void* param);
+
+    //Set the test function for the whole control.
+    void SetControlFunction(wxMaskedFunc* fn, void* param);
+
+    //Set all parameters at once, replacing old ones.
+    bool SetParams(const wxMaskedEditParams& params);
+
+    //Set the control's value from a undecorated text.
+    bool SetPlainValue(const wxString& plainValue);
+
+    //Sets the value for a field.
+    bool SetFieldValue(size_t index, const wxString& plainValue);
+
+    //Check characters, fields and the whole control's value.
+    long IsValid();
+
+    //Bell on error.
+    void SetBellOnError(bool bell)
+    {
+        m_bellOnError = bell;
+    }
+
+
+    //Accessors -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -  -
+    //Get the number of fields.
+    size_t GetFieldsCount() const
+    {
+        return m_arrPosS.size();
+    }
+
+    //Get control's value, without decorations.
+    wxString GetPlainValue() const;
+
+    //Get the current text in a field. Empty cells are filled with the padding
+    //char set for this field.
+    wxString GetFieldValue(size_t index) const;
+
+    //Get the mask. The string passed at SetMask()
+    wxString GetMask() const
+    {
+        return m_params.mask;
+    }
+
+    //Get the colours for different cases.
+    wxMaskedEditColours GetMaskedColours() const
+    {
+        return m_params.colours;
+    }
+
+    //Get the flags used in a field.
+    wxEditFieldFlags GetFieldFlags(size_t index) const
+    {
+        return m_params.fieldsFlags.at(index);
+    }
+
+    //Get the parameters.
+    wxMaskedEditParams GetParams() const
+    {
+        return m_params;
+    }
+
+    //Event handling
```

Comment:
Event handlers should be private.

Reply:
@vadz,
Making them private will not work, since they are bound n the different class and therefore I am getting a compiler error.

[quote]
c:\wxfork\include\wx\generic\maskededit.h(519): error C2248: 'wxMaskedEdit::OnKeyDown': cannot access private member declared in class 'wxMaskedEdit'
[/quote]

Testing on Win8.1 with MSVC 2017

For now I will make them public again.

Let me know what to do here..